### PR TITLE
Adds area checks to JavaScriptReplaceableUrl and resolves T4MVC/T4MVC#14

### PR DESCRIPTION
**Pull request by daveaglick**

### CODE REVIEW

File: T4MVCExtensions/T4Extensions.cs [hunk 1]
```diff
@@ -334,9 +334,11 @@ public static string JavaScriptReplaceableUrl(this UrlHelper urlHelper, ActionRe
...
 
             var specificActionUrl = (from r in RouteTable.Routes.OfType<Route>()
-                                     where r.DataTokens != null && r.Defaults != null
-                                     && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0
-                                         && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)
+                                     where r.Defaults != null
+                                     && ((string.IsNullOrEmpty(area) 
+                                        && (r.DataTokens == null || !r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0)))
+                                        || (r.DataTokens != null && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0 
+                                            && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)))
```
Comment (davidebbo) [inline T4MVCExtensions/T4Extensions.cs]: Sorry, my brain is struggling to make sense of these conditionals. :) What exactly is this doing in plain English? I think this code has reach a level of complexity where it would make sense to refactor some of the conditions into little well-named helper methods that would help readability.

Comment (somedave) [inline T4MVCExtensions/T4Extensions.cs]: Agree completely - I was just trying to keep to the original code style as much as possible. Now that I know you're amenable, I'll go ahead and refactor to make it clearer.

The code is essentially checking if there's an area specified in the `ActionResult` and if not, matching a route without `DataTokens` collection or without an "area" key; if so, matching a route that has a `DataTokens` collection with a matching "area" key.

Comment (davidebbo) [inline T4MVCExtensions/T4Extensions.cs]: Cool. Not blaming you for the messy code, as it grew that way over time as things usually do! If you have time to do a little refactoring, that would be welcome!

**Comment by daveaglick** [comment]
So this should be quite a bit clearer. Just let me know if you think it needs anything else.

**Comment by davidebbo** [comment]
Very nice, thanks! It's in 3.14.1.
